src/pykeen/triples/instances.py

```diff
@@ -21,20 +23,42 @@
     'MultimodalLCWAInstances',
 ]
 
+BatchType = TypeVar("BatchType")
+LCWASampleType = Tuple[MappedTriples, torch.FloatTensor]
+LCWABatchType = Tuple[MappedTriples, torch.FloatTensor]
+SLCWASampleType = TypeVar('SLCWASampleType', bound=MappedTriples)
+SLCWABatchType = Tuple[MappedTriples, MappedTriples, Optional[torch.BoolTensor]]
+
 
 @fix_dataclass_init_docs
 @dataclass
-class Instances(data.Dataset):
+class Instances(data.Dataset, Generic[BatchType], ABC):
     """Triples and mappings to their indices."""
 
     def __len__(self):  # noqa:D401
         """The number of instances."""
         raise NotImplementedError
 
+    def __getitem__(self, item: int) -> BatchType:  # noqa: D105
+        raise NotImplementedError
+
+    @classmethod
+    def from_triples(cls, mapped_triples: MappedTriples, num_entities: int) -> 'Instances':
+        """Create instances from mapped triples.
+
+        :param mapped_triples: shape: (num_triples, 3)
+            The ID-based triples.
+        :param num_entities:
+            The number of entities.
+        :return:
+            The instances.
+        """
+        raise NotImplementedError
+
 
 @fix_dataclass_init_docs
 @dataclass
-class SLCWAInstances(Instances):
+class SLCWAInstances(Instances[MappedTriples]):
```

Comment:
update to SLCWABatchType?

Reply:
Nope, is not ready for the future, since in the next PR the sLCWA batch type will change form